docs/en/guides/backend-profile.md

```diff
@@ -0,0 +1,44 @@
+# Thread dump merging mechanism
+The performance profile is an enhancement feature in the APM system. We are using the thread dump to estimate the method execution time, rather than adding many local spans. In this way, the resource cost would be much less than using distributed tracing to locate slow method. This feature is suitable in the production environment. This document introduces how thread dumps are merged into the final report as a stack tree(s).
+
+## Thread analyst
+### Read data and transform
+Read data from the database and convert it to a data structure in gRPC.
+```
+st=>start: Start
+e=>end: End
+op1=>operation: Load data using paging
+op2=>operation: Transform data using parallel
+
+st(right)->op1(right)->op2
+op2(right)->e
```

Comment:
I'd prefer to generate an image and put that image here, the source codes seems not that straightforward, same as others below

Reply:
Don't put the image in the source codes. It will increase the payload of source download and release. All documents keep in texts as much as possible. All images must be hosted at the website repo, and only links here. For these comments, we could make it better, but I still think, visualizing on 3rd party system is better.

Reply:
For this graph, @mrproliu , I think we should describe the logic of each step doing, rather than stream API concept.